docs/source/extensions.rst

```diff
@@ -0,0 +1,68 @@
+.. _extensions-catalog:
+
+********************************
+Extensions Catalog and Proposals
+********************************
```

Comment:
The content of the page looks good, but I propose a few changes to overall structure of the page (i.e., mainly just moving around some the content and changing section headings). Since these changes affect the whole page, it was tricky to propose these changes on the PR directly, so I created a separate PR that targets this PR to propose the changes https://github.com/NeurodataWithoutBorders/nwb-overview/pull/171  @rly please take a look at #171 and see what you think.